app/workers/mail_worker.rb

```diff
@@ -0,0 +1,68 @@
+class MailWorker
+  MAX_FAIL_COUNT = Settings.mail_worker.max_fail_count
+  PID_FILEPATH   = Settings.mail_worker.pid_filepath
+
+  def initialize
+    @q = MailService::Queue.new
```

Comment:
Not quite sure what the point of the MailService::Queue object is.  Surely all the worker needs to do is to find all the mails to send, then iterate through each one and process it.  I don't see what extra value the queue adds here - it's already queued in the database.